lib/include/tick/prox/prox_binarsity.h

```diff
@@ -4,63 +4,99 @@
-  using TProxWithGroups<T>::synchronize_proxs;
+  using TProxWithGroups<T, K>::proxs;
+  using TProxWithGroups<T, K>::is_synchronized;
+  using TProxWithGroups<T, K>::synchronize_proxs;
 
  public:
-  using TProxWithGroups<T>::get_class_name;
+  using TProxWithGroups<T, K>::get_class_name;
 
  protected:
-  std::unique_ptr<TProx<T> > build_prox(T strength, ulong start, ulong end,
-                                        bool positive) override;
+  std::unique_ptr<TProx<T, K> > build_prox(T strength, ulong start, ulong end,
+                                           bool positive) override;
 
  public:
   // This exists soley for cereal/swig
   TProxBinarsity() : TProxBinarsity(0, nullptr, nullptr, false) {}
 
   TProxBinarsity(T strength, SArrayULongPtr blocks_start,
                  SArrayULongPtr blocks_length, bool positive)
-      : TProxWithGroups<T>(strength, blocks_start, blocks_length, positive) {}
+      : TProxWithGroups<T, K>(strength, blocks_start, blocks_length, positive) {
+  }
 
   TProxBinarsity(T strength, SArrayULongPtr blocks_start,
                  SArrayULongPtr blocks_length, ulong start, ulong end,
                  bool positive)
-      : TProxWithGroups<T>(strength, blocks_start, blocks_length, start, end,
-                           positive) {}
+      : TProxWithGroups<T, K>(strength, blocks_start, blocks_length, start, end,
+                              positive) {}
+
+  // There's something odd on windows trying to copy the unique_ptr in the
+  // superclass
+  TProxBinarsity(const TProxBinarsity&) = delete;
+  TProxBinarsity(const TProxBinarsity&&) = delete;
+  TProxBinarsity& operator=(const TProxBinarsity&) = delete;
+  TProxBinarsity& operator=(const TProxBinarsity&&) = delete;
 
-  void call(const Array<T>& coeffs, T step, Array<T>& out, ulong start,
+  void call(const Array<K>& coeffs, T step, Array<K>& out, ulong start,
             ulong end) override;
 
   template <class Archive>
   void serialize(Archive& ar) {
-    ar(cereal::make_nvp("ProxWithGroups", cereal::base_class<TProx<T> >(this)));
+    ar(cereal::make_nvp("ProxWithGroups",
+                        cereal::base_class<TProx<T, K> >(this)));
   }
 
-  BoolStrReport compare(const TProxBinarsity<T>& that) {
+  BoolStrReport compare(const TProxBinarsity<T, K>& that) {
     std::stringstream ss;
     ss << get_class_name();
-    auto are_equal = TProxWithGroups<T>::compare(that, ss);
+    auto are_equal = TProxWithGroups<T, K>::compare(that, ss);
     return BoolStrReport(are_equal, ss.str());
   }
-  BoolStrReport operator==(const TProxBinarsity<T>& that) {
+  BoolStrReport operator==(const TProxBinarsity<T, K>& that) {
     return compare(that);
   }
 };
 
-using ProxBinarsity = TProxBinarsity<double>;
-using ProxBinarsityDouble = TProxBinarsity<double>;
-using ProxBinarsityFloat = TProxBinarsity<float>;
+template <class T, class K>
+std::unique_ptr<TProx<T, K> > TProxBinarsity<T, K>::build_prox(T strength,
```

Comment:
This needs to be in header because of `std::unique_ptr<TProx<T, K>`?

Reply:
templates should generally be in headers - windows was complaining too